File: java/src/com/frc4343/robot/commands/DriveWithJoystick.java
package com.frc4343.robot.commands;

import com.frc4343.robot.CommandBase;

public class DriveWithJoystick extends CommandBase {
    public DriveWithJoystick() {
        requires(chassis);
    }

    protected void initialize() {
    }

    protected void execute() {
        chassis.driveWithJoystick(oi.getJoystick());
    }

    protected boolean isFinished() {
        return false;
    }

    protected void end() {
    }

    protected void interrupted() {
    }
}

File: c++/src/Team4343/Commands/DriveWithJoystick.cpp
#include "DriveWithJoystick.h"

DriveWithJoystick::DriveWithJoystick() {
	Requires(chassis);
}

void DriveWithJoystick::Initialize() {
    previousXVal = 0.0f;
    previousYVal = 0.0f;
}

void DriveWithJoystick::Execute() {
	chassis->driveWithJoystick(oi->getJoystick(), previousXVal, previousYVal);
}

bool DriveWithJoystick::IsFinished() {
	return false;
}

void DriveWithJoystick::End() {
}

void DriveWithJoystick::Interrupted() {
}
